From f291a535faebbe6f8806f9c30715cdef6ff0a246 Mon Sep 17 00:00:00 2001
From: Andre Heider <a.heider@gmail.com>
Date: Mon, 21 Nov 2022 11:22:15 +0100
Subject: [PATCH 37/39] cli: print current HT mode

This was present in the ubus call provided by rpcd, but not the cli.

Signed-off-by: Andre Heider <a.heider@gmail.com>
---
 iwinfo_cli.c | 19 +++++++++++++++++--
 1 file changed, 17 insertions(+), 2 deletions(-)

diff --git a/iwinfo_cli.c b/iwinfo_cli.c
index 7aa990c..d70f7fb 100644
--- a/iwinfo_cli.c
+++ b/iwinfo_cli.c
@@ -533,6 +533,20 @@ static char * print_hwmodes(const struct iwinfo_ops *iw, const char *ifname)
 	return format_hwmodes(modes);
 }
 
+static const char *print_htmode(const struct iwinfo_ops *iw, const char *ifname)
+{
+	int mode;
+	const char *name;
+	if (iw->htmode(ifname, &mode))
+		mode = -1;
+
+	name = iwinfo_htmode_name(mode);
+	if (name)
+		return name;
+
+	return "unknown";
+}
+
 static char * print_mbssid_supp(const struct iwinfo_ops *iw, const char *ifname)
 {
 	int supp;
@@ -564,10 +578,11 @@ static void print_info(const struct iwinfo_ops *iw, const char *ifname)
 		print_ssid(iw, ifname));
 	printf("          Access Point: %s\n",
 		print_bssid(iw, ifname));
-	printf("          Mode: %s  Channel: %s (%s)\n",
+	printf("          Mode: %s  Channel: %s (%s)  HT Mode: %s\n",
 		print_mode(iw, ifname),
 		print_channel(iw, ifname),
-		print_frequency(iw, ifname));
+		print_frequency(iw, ifname),
+		print_htmode(iw, ifname));
 	if (iw->center_chan1 != NULL) {
 		printf("          Center Channel 1: %s",
 			print_center_chan1(iw, ifname));
-- 
2.35.1

